### ISSUE TITLE
Multiple screens, scaling/DPI and issues with WinForms DPI awereness with SkiaSharp

### ISSUE THREAD

**Issue by jensbrak** [mono/SkiaSharp#1553]
Hello,

TL;DR: Windows Forms recently new API call to set DPI-awereness, nowadays included in the standard VS template, affects SkiaSharp. To me, in an unexpected way that I just try to understand. I seek advice or some enlightment, not nescessarly solutions.

The longer story:

However, beyond that, there are several challenges I have related to the very concept of my application. They are present regardless on how SkiaSharp work so I'll take them last.  I noticed that creating a project using VS Template for WinForms added a new line in `Program.cs`. It now looks like this out of the box:

```csharp

    static class Program
    {
        /// <summary>
        ///  The main entry point for the application.
        /// </summary>
        [STAThread]
        static void Main()
        {
            Application.SetHighDpiMode(HighDpiMode.SystemAware);
            Application.EnableVisualStyles();
            Application.SetCompatibleTextRenderingDefault(false);
            Application.Run(new PhiddleForm());
        }
    }
```

The firt line is new. This API call was introduced a year ago or so and previously it was needed to alter manifest to affect DPI mode. The reason I am here mentioning this at all, is that that line is what made my SkiaSharp render properly. Or at least I have narrowed it down to the fact that changing or even removing that line affects how SkiaSharp render.

What seems to be affecting the behaviour too is using multiple screens. Which is not surprising since it's very likely that multiple screens might have different resolutions. Which in turn is rather significant when playing with DPI, scaling etc. What happens is that my transparency I use (by combining SkiaSharp and Windows Forms in this case) stops being transparent and I need to understand more about what's happening.

I do not rule out that there might be a need for Skia or SkiaSharp to adress something but I honestly don't know. Still, I use SkiaSharp and I use it to make a Desktop app for Windows and Mac. The application implements a screen ruler-kind of functionality. Which relies on making it possible to show the desktop and whatever applications/windows that are "beneath" my app. This is where things get tricky and this is the reason I use WinForms component of SkiaSharp at all. Couldn't get it to work with other variants (Xamarin, UPF).

Since I want to measure screen pixels a basic question will arise: what exactly do I want to measure? In the world of physical resolutions (and if you ask, most people will answer): pixels. But what if the screen use scaling? What do I want to measure then? The actual pixels or the scaled pixels? And to complicate it, if multiple screens and different resolutions/scaling is present - what is right or wrong? It quickly becomes tricky - not only to get it to work but to have a view on what's desirable and not.

To me, I think I want actual pixels, no matter screen or resolution. Or maybe, if possible, be able to offer the choice of how to measure but initial testing indicates that this is quite complex if not impossible. Especially since I have two platforms I want it to work on. Why settle for easy things? :) 

Right now my app works just fine - at least in the single monitor scenario (I dodged the complexity of multiple screens). Or it did work until I upgraded to .NET 5. I now know how to restore my transparency (by removing or altering the mentioned API call) and get my SkiaSharp app to work as before.

However, I am feeling rather lost about how to adress this further. In fact, not sure if I am at the right place. Still, many of you use SkiaSharp in various ways and maybe there are some of you with more knowledge about multiple screens, resolutions, scaling and how Skia render - that might help me understand better. Or just give up ;)
